Brightspace/insights-engagement-dashboard
# Profile Images for User Drill - US122293

# User Profile Image in the User Drill

## Implementation Notes 

**Profile Image**

![](https://i.ibb.co/CWK4w3T/with-profile.png)

**User had no image or token is invalid (initials)** 

![](https://i.ibb.co/RHq5J4F/with-initials.png)

**Profile image is loading or bad UserID**

![](https://i.ibb.co/g4qNfxP/loading.png)


## Quality Notes
### Functional Testing
  - Unit Test
    - Give the profile component a fake user entity and check if the component reflects the right name and last name
  - Manual Test
    - Check that image appears with proper token and user id
    - check that initials appear with no profile image, with proper token and id
    - check that initials appear with profile image but invalid token, with valid user id
    - check that user placeholder appears while fetching user entity
    - check that user placeholder appears if user id is invalid

### Security
  - Fetches the oauth token using D2L.LP.Oauth, same as is done on the front page.
### UX
  - requires demo to Kevin  

### Rest N/A

FYI @johngwilkinson @Vitalii-Misechko @rohitvinnakota @MykolaGalian @rohitvinnakota
